src/main/java/reactor/netty/http/server/WebsocketServerOperations.java

```diff
@@ -93,7 +93,7 @@
 			       .set(replaced.nettyRequest.headers());
 
 			if (channel().pipeline()
-			             .get(NettyPipeline.CompressionHandler) != null) {
+			             .get(NettyPipeline.CompressionHandler) != null || compressionPredicate != null) {
```

Comment:
Shouldn't the predicate be tested like in `HttpServerOperations#preSendHeadersAndStatus`?

```java
if (channel().pipeline().get(NettyPipeline.CompressionHandler) != null ||
		(compressionPredicate != null && compressionPredicate.test(this, this))) {
    // ...
}
```

Reply:
We cannot test the predicate as it makes sense only for HTTP.
We use just the existence of the predicate as an indicator that the user wants `compression` on the server.
I would never use `compress(2048)` when I use the HTTP client for WebSocket, but when you are on the spring gateway where the HTTP client serves both HTTP and WebSocket it make sense.